gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/query/HiveAvroORCQueryGenerator.java

```diff
@@ -824,6 +828,33 @@ public static String generateCleanupDDL(String stagingTableName, Optional<String
     return String.format("DROP TABLE IF EXISTS `%s`.`%s`", stagingDbName, stagingTableName) + "\n";
   }
 
+  /**
+   * Generate DDL for dropping partitions of a table.
+   * <p>
+   * ALTER TABLE finalTableName DROP IF EXISTS PARTITION partition_spec, PARTITION partition_spec, ...;
+   * </p>
+   * @param finalTableName Table name where partitions are dropped
+   * @param partitionDMLInfos list of Partition to be dropped
+   * @return DDL to drop partitions in <code>finalTableName</code>
+   */
+  public static String generateDropPartitionsDDL(final String finalTableName, final List<Map<String, String>> partitionDMLInfos, boolean purge) {
+
+    if (partitionDMLInfos.isEmpty()) {
+      return StringUtils.EMPTY;
+    }
+
+    StringBuilder ddl = new StringBuilder(String.format("ALTER TABLE %s DROP IF EXISTS ", finalTableName));
+
+    // Join the partition specs
+    ddl.append(Joiner.on(", ").join(Iterables.transform(partitionDMLInfos, PARTITION_SPEC_GENERATOR)));
+
+    if (purge) {
+      ddl.append(" PURGE");
```

Comment:
Purge is not available in 0.13, also it only purges data for managed Hive tables not external. So we need to manually cleanup data via FS apis